temporalio/exceptions.py

```diff
@@ -1,4 +1,22 @@
-"""Common Temporal exceptions."""
+"""Common Temporal exceptions.
+
+# Temporal Failure
+
+Most Temporal SDKs have a base class that the other Failures extend.
+In python, it is the ``FailureError``.
+
+# Application Failure
+
+Workflow, and Activity, and Nexus Operation code use Application Failures to
+communicate application-specific failures that happen.
+This is the only type of Temporal Failure created and thrown by user code.
+In the Python SDK, it is the ``ApplicationError``.
+
+# References
+
+More information can be found in the docs at
+https://docs.temporal.io/references/failures#workflow-execution-failures.
+"""
```

Comment:
I think maybe let's not make these additions: We don't reference or explain other SDKs in the Python SDK codebase, Nexus hasn't been added to the SDK yet, and the URL would be  more helpful under ApplicationError and or FailureError docstrings.